bayesian_cnn_prometheus/preprocessing/data_generator.py

```diff
@@ -74,17 +75,21 @@ def _generate_chunks(data_subset: np.ndarray, chunk_size: tuple = (32, 32, 16),
         :param stride: three-elements tuple with steps value to make in each axis
         :return: generator which produces chunks with size (a, b, c)
         """
+        for _ in range(self.chunks_number):
+            chunk_x, chunk_y, chunk_z = chunk_size
+            x, y, z = self._get_random_chunk_coords(data_subset, stride)
+            chunk = data_subset[x:x + chunk_x, y:y + chunk_y, z:z + chunk_z]
+            if chunk.shape == tuple(chunk_size):
+                yield chunk
+
+    @staticmethod
+    def _get_random_chunk_coords(data_subset, stride):
         origin_x, origin_y, origin_z = data_subset.shape
-        chunk_x, chunk_y, chunk_z = chunk_size
         stride_x, stride_y, stride_z = stride
-
-        for x in range(0, origin_x, stride_x)[:-1]:
-            for y in range(0, origin_y, stride_y)[:-1]:
-                for z in range(0, origin_z, stride_z)[:-1]:
-                    chunk = data_subset[x:x + chunk_x,
-                                        y:y + chunk_y, z:z + chunk_z]
-                    if chunk.shape == tuple(chunk_size):
-                        yield chunk
+        x = random.sample(range(0, origin_x, stride_x)[1:-1], k=1)[0]
+        y = random.sample(range(0, origin_y, stride_y)[1:-1], k=1)[0]
+        z = random.sample(range(0, origin_z, stride_z)[1:-1], k=1)[0]
```

Comment:
Spoko, tylko ja bym zrobił `_get_random_chunk_coords` generatorem który zwracałby wszystkie możliwe kombinacje współrzędnych wygenerowanych w losowy sposób

Czyli coś w stylu:

```
x = random.sample(range(0, origin_x, stride_x, k=len(x))
...
for coords in product(x,y,z): #product is the same as a nested for loop
   yield coords
```
![obraz](https://user-images.githubusercontent.com/21079319/144226179-d28b0b7f-0f18-43f9-ba2b-eba5ccc23f26.png)

Zalety które widzę to:
1. my dalej przechodzimy przez wszystkie chunki z obrazka, a nie tylko przez tę liczbę którą chcieliśmy podać
2. chunki się nie powtarzają. Teraz mogą z racji tego iż za każdym razem losujemy od nowa

Reply:
A tak z innej beczki, to proponuję pozamieniać wszystkie zagnieżdżone `for` w naszym projekcie na `product` bo będzie mniej wcięć :D

Reply:
Dzięki za komentarz, to super pomysł :)